File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*; 
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Deque<Integer>q=new ArrayDeque<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                q.add(1);
            }
            else{
                if(q.isEmpty())
                    return false;
                q.pollFirst();
            }
        }
        if(!q.isEmpty())
            return false;

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    answer = True
    q=deque()
    for i in s : 
        if i=='(':
            q.append('(')
        else:
            if len(q)==0:
                return False
            else:
                q.popleft()
    if len(q)>0:
        return False
    return answer